crates/sovereign-sdk/module-system/sov-modules-stf-blueprint/src/lib.rs

```diff
@@ -459,35 +472,36 @@ where
                     batch_workspace,
                 );
 
-                let (batch_receipt, checkpoint) = self.end_soft_confirmation(
+                match self.end_soft_confirmation(
                     current_spec,
                     sequencer_public_key,
                     soft_confirmation,
                     tx_receipts,
                     batch_workspace,
-                );
-
-                self.finalize_soft_confirmation(
-                    current_spec,
-                    batch_receipt,
-                    checkpoint,
-                    pre_state,
-                    soft_confirmation,
-                )
+                ) {
+                    (Ok(batch_receipt), checkpoint) => Ok(self.finalize_soft_confirmation(
+                        current_spec,
+                        batch_receipt,
+                        checkpoint,
+                        pre_state,
+                        soft_confirmation,
+                    )),
+                    (Err(err), _checkpoint) => {
```

Comment:
Do we ever use checkpoint in case of Err? If not, there is no reason to return it

Reply:
https://github.com/chainwayxyz/citrea/issues/1047